Tasks/Maven/maventask.ts

```diff
@@ -203,6 +203,55 @@ function createMavenSQRunner(sqHostUrl, sqHostUsername, sqHostPassword, sqDbUrl?
     return mvnsq;
 }
 
+function processMavenOutput(data) {
+    data = data.toString();
+    var input = data;
+    var severity = 'NONE';
+    if(data.substring(0, 1) == '[') {
+        var rightIndex = data.indexOf(']');
+        if(rightIndex > 0) {
+            severity = data.substring(1, rightIndex);
+
+            if(severity == 'ERROR' || severity == 'WARNING') {
+                // Try to match output like
+                // /Users/user/agent/_work/4/s/project/src/main/java/com/contoso/billingservice/file.java:[linenumber, columnnumber] error message here
+                // A successful match will return an array of 5 strings - full matched string, file path, line number, column number, error message
+                input = input.substring(rightIndex + 1);
+                var compileErrorsRegex = /([a-zA-Z0-9_ \-\/.]+):\[([0-9]+),([0-9]+)\](.*)/g;
+                var matches = [];
+                var match;
+                while (match = compileErrorsRegex.exec(input.toString())) {
+                    matches = matches.concat(match);
+                }
+
+                if(matches != null) {
+                    var index = 0;
+                    while (index + 4 < matches.length) {
+                        tl.debug('full match = ' + matches[index + 0]);
+                        tl.debug('file path = ' + matches[index + 1]);
+                        tl.debug('line number = ' + matches[index + 2]);
+                        tl.debug('column number = ' + matches[index + 3]);
+                        tl.debug('message = ' + matches[index + 4]);
+                        var messageToLog = matches[index + 1] + '(' + matches[index + 2] + ',' + matches[index + 3] +
+                            '): ' + severity.toLowerCase() + ' : ' + matches[index + 4];
```

Comment:
`+ severity.toLowerCase() +`

The windows agent doesn't seem to output severity, do you think we should match the output?  Everything listed is considered an "error", so this field may not add much value.